lib/internal/Magento/Framework/DB/Ddl/Table.php

```diff
@@ -427,13 +428,23 @@ public function addForeignKey($fkName, $column, $refTable, $refColumn, $onDelete
             default:
                 $onDelete = self::ACTION_NO_ACTION;
         }
+        switch ($onUpdate) {
+            case self::ACTION_CASCADE:
+            case self::ACTION_RESTRICT:
+            case self::ACTION_SET_DEFAULT:
+            case self::ACTION_SET_NULL:
+                break;
+            default:
+                $onUpdate = self::ACTION_NO_ACTION;
+        }
 
         $this->_foreignKeys[$upperName] = [
             'FK_NAME' => $fkName,
             'COLUMN_NAME' => $column,
             'REF_TABLE_NAME' => $refTable,
             'REF_COLUMN_NAME' => $refColumn,
-            'ON_DELETE' => $onDelete
+            'ON_DELETE' => $onDelete,
+            'ON_UPDATE' => $onUpdate
```

Comment:
@xi-ao Thank you for your contribution. Unfortunately, this is not full support of this feature. 
Please take a look at [/lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php#L2236-L2256](https://github.com/magento/magento2/blob/develop/lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php#L2236-L2256) and [lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php#L1699-L1714](https://github.com/magento/magento2/blob/develop/lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php#L1699-L1714)
This feature requires much more work. In order to make sure that it works, it would be great to cover this functionality with integration test